xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-livetable/src/test/java/org/xwiki/livedata/internal/livetable/LiveTableLiveDataPropertyStoreTest.java

```diff
@@ -163,25 +169,80 @@ void getAll() throws Exception
+        descriptor2.setDescription("The computed total amount.");
+        descriptor2.setType("Computed");
+        LiveDataPropertyDescriptor.DisplayerDescriptor displayer2 =
+            new LiveDataPropertyDescriptor.DisplayerDescriptor();
+        displayer2.setId("xObjectProperty");
+        descriptor2.setDisplayer(displayer2);
+
+        LiveDataPropertyDescriptor descriptor3 = new LiveDataPropertyDescriptor();
+        descriptor3.setId("status");
+        descriptor3.setName("Status");
+        descriptor3.setDescription("The status.");
+        descriptor3.setType("List");
+        descriptor3.setSortable(false);
+        LiveDataPropertyDescriptor.DisplayerDescriptor displayer3 =
+            new LiveDataPropertyDescriptor.DisplayerDescriptor();
+        displayer3.setId("xObjectProperty");
+        descriptor3.setDisplayer(displayer3);
+        LiveDataPropertyDescriptor.FilterDescriptor filter3 = new LiveDataPropertyDescriptor.FilterDescriptor();
+        filter3.setId("list");
+        LiveDataPropertyDescriptor.OperatorDescriptor operator1 = new LiveDataPropertyDescriptor.OperatorDescriptor();
+        operator1.setId("empty");
+        LiveDataPropertyDescriptor.OperatorDescriptor operator2 = new LiveDataPropertyDescriptor.OperatorDescriptor();
+        operator2.setId("equals");
+        filter3.setOperators(List.of(operator1, operator2));
+        filter3.setParameter("searchURL",
+            "/xwiki/rest/wikis/wiki/classes/Some.Class/properties/status/values?fp={encodedQuery}");
+        descriptor3.setFilter(filter3);
+
+        LiveDataPropertyDescriptor descriptor4 = new LiveDataPropertyDescriptor();
+        descriptor4.setId("levels");
+        descriptor4.setType("Levels");
+        LiveDataPropertyDescriptor.DisplayerDescriptor displayer4 =
+            new LiveDataPropertyDescriptor.DisplayerDescriptor();
+        displayer4.setId("xObjectProperty");
+        descriptor4.setDisplayer(displayer4);
+        LiveDataPropertyDescriptor.FilterDescriptor filter4 = new LiveDataPropertyDescriptor.FilterDescriptor();
+        filter4.setId("list");
+        LiveDataPropertyDescriptor.OperatorDescriptor operatorEmpty =
+            new LiveDataPropertyDescriptor.OperatorDescriptor();
+        operatorEmpty.setId("empty");
+        filter4.setOperators(List.of(operatorEmpty));
+        filter4.setParameter("options", List.of(
+            Map.of("value", "edit", "label", "Edit right"),
+            Map.of("value", "delete", "label", "delete")
+        ));
+        descriptor4.setFilter(filter4);
         Collection<LiveDataPropertyDescriptor> properties = this.propertyStore.get();
 
-        String expectedJSON =
-            "[" + getExpectedDocPropsJSON() + "," + expectedClassProps.toString().replace('\'', '"') + "]";
-        assertEquals(expectedJSON, this.objectMapper.writeValueAsString(properties));
+        assertEquals(List.of(descriptor0, descriptor1, descriptor2, descriptor3, descriptor4), properties);
```

Comment:
As discussed on the chat, I've refactored the test to use objects instead of JSON serialization to avoid issues with order of parameters.

Reply:
Thanks for refactoring this!